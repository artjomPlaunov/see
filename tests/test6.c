int add(int a, int b) {
  int c = 0;
}

int main () {
  int a = 1, b = 2;
  int f[2] = {0,0};
  int c = 3;
  int d[4] = {4,5,6,7};
  int s = 0;
  int j = 0;
}
