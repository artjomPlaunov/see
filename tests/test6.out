(Ast.Prog
   [(Ast.Fun (Ast.INT_T,
       (Ast.ParamLst
          [(Ast.Param (Ast.INT_T, (Ast.Ident "a")));
            (Ast.Param (Ast.INT_T, (Ast.Ident "b")))]),
       (Ast.Ident "add"),
       (Ast.Block
          [(Ast.Stm_declLst (Ast.INT_T, [(Ast.Decl ((Ast.Ident "c"), 0))]))])
       ));
     (Ast.Fun (Ast.INT_T, (Ast.ParamLst []), (Ast.Ident "main"),
        (Ast.Block
           [(Ast.Stm_declLst (Ast.INT_T,
               [(Ast.Decl ((Ast.Ident "a"), 1));
                 (Ast.Decl ((Ast.Ident "b"), 2))]
               ));
             (Ast.Stm_arrayDecl ((Ast.Ident "f"), 2, [0; 0]));
             (Ast.Stm_declLst (Ast.INT_T, [(Ast.Decl ((Ast.Ident "c"), 3))]));
             (Ast.Stm_arrayDecl ((Ast.Ident "d"), 4, [4; 5; 6; 7]));
             (Ast.Stm_declLst (Ast.INT_T, [(Ast.Decl ((Ast.Ident "s"), 0))]));
             (Ast.Stm_declLst (Ast.INT_T, [(Ast.Decl ((Ast.Ident "j"), 0))]))
             ])
        ))
     ])